### ISSUE TITLE
ERROR: could not find IP range for mask ffff0000 during network create

### ISSUE THREAD

**Issue by sflxn** [vmware/vic#4157]
During docker-compose testing, I got the following error:
```
ERROR: could not find IP range for mask ffff0000
```

This was after a battery of compatibility testing.

VCH installation command:
```
./vic-machine-linux create --target=192.168.218.207 --image-store=datastore1 --name=vic-docker --user=root --password=xxxx --compute-resource="/ha-datacenter/host/esxbox.localdomain/Resources" --bridge-network=vic-network --debug=0 --no-tls --volume-store=datastore1/test:default --force
```

Reinstalling the VCH fixes this.  Container logs attached.
[container-logs (1).zip](https://github.com/vmware/vic/files/822967/container-logs.1.zip)

**Comment by hmahmood** [comment]
Logs indicate that `context.NewScope` (which creates a bridge network) is getting called 17 times, i.e. the caller is attempting to create more than 16 bridge networks. This is not currently supported.

**Comment by sflxn** [comment]
I have 5 compose yml files.  To reproduce,

1. Download docker-compose 1.9.0 and 1.11.2
2. Run docker-compose -f <file 1> up -d
3. Run docker-compose -f <file 1> down
4. Repeat 2 and 3 for file 2, 3, 4, and 5
5. Repeat 2-4 for each version of compose binary
6. Repeat again until this issue appears.  From my test, after performing these tests for each version of the compose binary once, the issue would appear within the next 1-3 compose up instance.

[compose.tar.gz](https://github.com/vmware/vic/files/823023/compose.tar.gz)

**Comment by sflxn** [comment]
I've just verified that once I get in this state, both docker compose and docker network create fails with that error.

**Comment by anchal-agrawal** [comment]
Need input from @hmahmood to estimate and prioritize this, if this is indeed a bug. If `docker-compose down` isn't working as expected, we should update the issue title.

**Comment by hmahmood** [comment]
So currently we have the default "width" of the bridge networks as hardcoded in the code to /16. We currently have an option in vic-machine to specify the bridge network pool (`--bridge-network-range`), but the individual networks will still be /16. We can add another option to let the user override that to say, /24, which would allow for more than 16 networks, depending on the pool size for bridge networks.

**Comment by mdubya66** [comment]
Closed #5694 as a dupe

**Comment by mdubya66** [comment]
Marking high per #5694.

**Comment by corrieb** [comment]
If it were the case that we could only have a maximum of 16 networks at any one time, that wouldn't be an unreasonable limitation. What I highlighted in #5694 is far more serious than that. It states that you can only EVER have 16 networks. If you create and delete 16 times, the VCH becomes completely unusable. You cannot create a network at that point. docker-compose does clean up after itself and it exhibits exactly this behavior. So this is an issue about being able to re-use IP ranges that are no longer in use.

**Comment by corrieb** [comment]
Unless there is a workaround to this that does not involve uninstalling the VCH, I disagree with the reduction of priority of this issue.

**Comment by hmahmood** [comment]
I agree with @corrieb about the severity of this; we should fix this for 1.2.
